File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    current_nums = [n for n in lottos if n in win_nums]
    zero_cnt = lottos.count(0)
    
    min_match = len(current_nums)
    max_match = min_match + zero_cnt
    
    def get_rank(cnt):
        return 7 - cnt if cnt >= 2 else 6
    
    return [get_rank(max_match), get_rank(min_match)]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    let currentNums = lottos.filter(n => win_nums.includes(n));
    let zeroCnt = lottos.reduce((acc, cur) => cur === 0 ? acc + 1 : acc, 0);
    
    const minMatch = currentNums.length;
    const maxMatch = minMatch + zeroCnt;
    const getRank = cnt => cnt >= 2 ? 7 - cnt : 6;
    return [getRank(maxMatch), getRank(minMatch)];
}